server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -247,10 +287,33 @@ public SnapshotsService(
      * @param listener snapshot completion listener
      */
     public void executeSnapshot(final CreateSnapshotRequest request, final ActionListener<SnapshotInfo> listener) {
-        createSnapshot(
-            request,
-            ActionListener.wrap(snapshot -> addListener(snapshot, ActionListener.map(listener, Tuple::v2)), listener::onFailure)
-        );
+        if (isCentralSnap) {
+            createSnapshotCentralized(
+                request,
+                ActionListener.wrap(snapshot -> addListener(snapshot, ActionListener.map(listener, Tuple::v2)), listener::onFailure)
+            );
+        } else {
+            createSnapshot(
+                request,
+                ActionListener.wrap(snapshot -> addListener(snapshot, ActionListener.map(listener, Tuple::v2)), listener::onFailure)
+            );
+        }
+    }
+
+    public void createSnapshotValidations(final CreateSnapshotRequest request, final ActionListener<Snapshot> listener) {
+        final String repositoryName = request.repository();
+        final String snapshotName = indexNameExpressionResolver.resolveDateMathExpression(request.snapshot());
+        validate(repositoryName, snapshotName);
+        // TODO: create snapshot UUID in CreateSnapshotRequest and make this operation idempotent to cleanly deal with transport layer
```

Comment:
Couple of questions:
1. If this is not idempotent then your in-memory map of maintaining status of create snapshot operations is also not serving any purpose now. Is this correct?
2. After you create UUIDs at REST layer, in what scenarios do you see create snapshot being invoked multiple times for same UUIDs?